abc-bank/Customers/Customer.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+using abc_bank.Accounts;
+using abc_bank.Reports;
+using abc_bank.Transactions;
+using abc_bank.Utilities;
+
+namespace abc_bank.Customers {
+  public class Customer {
+
+    #region Properties  
+
+    public int Id { get; } = -1;
+
+    public string Name { get; }
+
+    public List<IAccount> Accounts { get; protected set; } = new List<IAccount>();
+
+    public int NumberOfAccounts => Accounts.Count;
+
+    public string Statement => Statements.CustomerStatement(this);
+
+    public double TotalInterestEarned {
+      get {
+        var total = 0.0;
+
+        foreach (IAccount account in Accounts) {
+          total += account.InterestEarned;
+        }
+
+        return total;
+      }
+    }
+
+    #endregion
+
+    #region CTOR
+
+    public Customer(string name, int customerId) {
+      Id = customerId;
+      Name = name;
+    }
+
+    #endregion
+
+    #region Public Methods
+
+    public Customer OpenAccount(IAccount account) {
```

Comment:
Should have a check to ensure the account and/or accountId doesn't already exist in the Accounts List.  Noticed and thought of that during pre-commit review.  Noting here is sufficient for this purpose however.

Also, in my humble opinion, Accounts and Transactions both should reside in the Bank class and Customer/Account should only maintain a list of AccountIds, and Account class a list of TransactionIds - but well beyond scope.